Tests/Functional/Consumers/AsyncQueueConsumerTest.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+namespace Smartbox\Integration\FrameworkBundle\Tests\Unit\Consumers;
+
+use JMS\Serializer\SerializerInterface;
+use PhpAmqpLib\Message\AMQPMessage;
+use PhpAmqpLib\Wire\AMQPTable;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\AsyncQueueConsumer;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\AsyncQueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueMessage;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\QueueProtocol;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Tools\Helper\SmartesbHelper;
+
+/**
+ * Class AsyncQueueConsumerTest
+ * @group async-queue-consumer
+ */
+class AsyncQueueConsumerTest extends TestCase
```

Comment:
Is this test functional or unit test? Because I see all the dependencies mocked

Reply:
Pending to review this test class until discuss the previous comment

Reply:
Well to be functional we'd need to include the driver so it makes sense to consider this as unit testing.

Reply:
Then we need to change the folder where the test is because at the moment it is placed under `Tests/Functional/Consumers` and the namespace is `Smartbox\Integration\FrameworkBundle\Tests\Unit\Consumers`

Reply:
@andreskrey ⬆️

Reply:
File moved to /Tests/Unit/Consumer and namespace aligned